seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/checkpoint/CheckpointCoordinator.java

```diff
@@ -397,7 +397,7 @@ private void startTriggerPendingCheckpoint(
                     LOG.info("wait checkpoint completed: " + pendingCheckpoint.getCheckpointId());
                     PassiveCompletableFuture<CompletedCheckpoint> completableFuture =
                             pendingCheckpoint.getCompletableFuture();
-                    completableFuture.whenComplete(
+                    completableFuture.whenCompleteAsync(
```

Comment:
Note: make sure the `TaskAcknowledgeOperation` thread will be release after operation called.